prometheus/metric_test.go

```diff
@@ -90,3 +93,263 @@ func TestWithExemplarsMetric(t *testing.T) {
+				PositiveBuckets: map[int]int64{
+					0: 1, 2: 2, 4: 2,
+				},
+				NegativeBuckets: map[int]int64{},
+				ZeroBucket:      1,
+
+				NativeHistogramSchema:        2,
+				NativeHistogramZeroThreshold: 2.938735877055719e-39,
+				CreatedTimestamp:             now,
+				Bucket: []*dto.Bucket{
+					{
+						CumulativeCount: PointOf(uint64(6)),
+						UpperBound:      PointOf(float64(1)),
+					},
+					{
+						CumulativeCount: PointOf(uint64(8)),
+						UpperBound:      PointOf(float64(2)),
+					},
+					{
+						CumulativeCount: PointOf(uint64(11)),
+						UpperBound:      PointOf(float64(5)),
+					},
+					{
+						CumulativeCount: PointOf(uint64(13)),
+						UpperBound:      PointOf(float64(10)),
+					},
+				},
+				Exemplars: []Exemplar{
+					{
+						Timestamp: now,
+						Value:     10,
+					},
+				},
+				Want: &dto.Metric{
+					Histogram: &dto.Histogram{
+						SampleCount:   proto.Uint64(6),
+						SampleSum:     proto.Float64(7.4),
+						Schema:        proto.Int32(2),
+						ZeroThreshold: proto.Float64(2.938735877055719e-39),
+						ZeroCount:     proto.Uint64(1),
+						PositiveSpan: []*dto.BucketSpan{
+							{Offset: proto.Int32(0), Length: proto.Uint32(5)},
+						},
+						PositiveDelta: []int64{1, -1, 2, -2, 2},
+						Exemplars: []*dto.Exemplar{
+							{
+								Value:     PointOf(float64(10)),
+								Timestamp: timestamppb.New(now),
+							},
+						},
+						Bucket: []*dto.Bucket{
+							{
+								CumulativeCount: PointOf(uint64(6)),
+								UpperBound:      PointOf(float64(1)),
+							},
+							{
+								CumulativeCount: PointOf(uint64(8)),
+								UpperBound:      PointOf(float64(2)),
+							},
+							{
+								CumulativeCount: PointOf(uint64(11)),
+								UpperBound:      PointOf(float64(5)),
+							},
+							{
+								CumulativeCount: PointOf(uint64(13)),
+								UpperBound:      PointOf(float64(10)),
+								Exemplar: &dto.Exemplar{
+									Timestamp: timestamppb.New(now),
+									Value:     PointOf(float64(10)),
+								},
+							},
+						},
+						CreatedTimestamp: timestamppb.New(now),
+					},
+				},
+			},
+		}
+
+		for _, tc := range tcs {
+			m, err := NewDummyConstNativeHistogram(NewDesc(tc.Name, "None", []string{}, map[string]string{}), tc.Count, tc.Sum, tc.PositiveBuckets, tc.NegativeBuckets, tc.ZeroBucket, tc.NativeHistogramSchema, tc.NativeHistogramZeroThreshold, tc.CreatedTimestamp, tc.Bucket)
+			if err != nil {
+				t.Fail()
+			}
+			metricWithExemplar, err := NewMetricWithExemplars(m, tc.Exemplars[0])
+			if err != nil {
+				t.Fail()
+			}
+			got := &dto.Metric{}
+			err = metricWithExemplar.Write(got)
+			if err != nil {
+				t.Fail()
+			}
+
+			if !proto.Equal(tc.Want, got) {
+				t.Errorf("want histogram %q, got %q", tc.Want, got)
+			}
+
+		}
+	})
+}
+
+func PointOf[T any](value T) *T {
+	return &value
+}
+
+// This is a dummy Histogram which has both buckets and nativehistogram
+// to test metrics with exemplars
+func NewDummyConstNativeHistogram(
```

Comment:
Let's not export this, so that nobody will use this for real. Also, how about a more descriptive name like "newNativeHistogramWithClassicBuckets"? And finally, end a sentence with a period. :)

```suggestion
// newNativeHistogramWithClassicBuckets returns a Metric representing
// a native histogram that also has classic buckets. This is for testing purposes.
func newNativeHistogramWithClassicBuckets(
```